packages/components/src/components/amount-input/amount-input.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from 'react';
+import Input from '../input';
+
+type TAmountInput = {
+    currency: string;
+    decimalPoints?: number;
+    disabled?: boolean;
+    initialValue?: number;
+    label?: React.ReactNode;
+    maxDigits: number;
+};
+
+const AmountInput = ({
+    currency,
+    decimalPoints = 2,
+    disabled = false,
+    initialValue = 0,
+    label,
+    maxDigits,
+}: TAmountInput) => {
+    const [value, setValue] = useState(initialValue);
+    const [focus, setFocus] = useState(false);
+
+    const displayNumber = (number: number) => number.toLocaleString('en-US', { minimumFractionDigits: decimalPoints });
+
+    return (
+        <div className='amount-input-wrapper'>
+            <span>{label}</span>
+            <div className='amount-input-container'>
+                <Input
+                    className='amount-input'
+                    disabled={disabled || focus}
+                    type='text'
+                    value={`${displayNumber(value)} ${currency}`}
+                />
+                <Input
+                    className='amount-input'
+                    data-testid='dt_amount-input'
+                    disabled={disabled}
+                    max_characters={displayNumber(Math.pow(10, maxDigits - 1) / decimalPoints).length}
```

Comment:
@rostislav-deriv Would you please elaborate on why you are doing this? 🤔 Using the `maxDigits` directly wouldn't work? 🤔

Reply:
Yes, you are right, I think this is redundant here since the goal of it is achieved in the `onChangeHandler`